chain/beacon/sync_manager.go

```diff
@@ -122,12 +120,12 @@ func (s *SyncManager) Run() {
 	lastRoundTime := 0
 	// the context being used by the current sync process
 	var lastCtx context.Context
-	_, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
If we start using this context, I feel like we might want to use the same `ctx` for `go s.Sync` below, no?